gloo/docs/gloo_routing/virtual_services/security/ldap/_index.md

```diff
@@ -0,0 +1,422 @@
+If you are using Open Source Gloo, this tutorial will not work.
+{{% /notice %}}
+
+{{< readfile file="/static/content/setup_notes" markdown="true">}}
+
+
+#### Create a simple Virtual Service
+Let's start by creating a simple service (an `Upstream`, in Gloo terminology) that will return "Hello World" when 
+receiving HTTP requests:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: http-echo
+  name: http-echo
+spec:
+  selector:
+    matchLabels:
+      app: http-echo
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: http-echo
+    spec:
+      containers:
+      - image: hashicorp/http-echo:latest
+        name: http-echo
+        args: ["-text='Hello World!'"]
+        ports:
+        - containerPort: 5678
+          name: http
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: http-echo
+  labels:
+    service: http-echo
+spec:
+  ports:
+  - port: 5678
+    protocol: TCP
+  selector:
+    app: http-echo
+EOF
+```
+
+Now we can create a Virtual Service that routes any requests with the `/echo` prefix to the `echo` service.
+
+{{< tabs >}}
+{{< tab name="yaml" codelang="yaml">}}
+{{< readfile file="gloo_routing/virtual_services/security/ldap/vs-echo-no-auth.yaml">}}
+{{< /tab >}}
+{{< tab name="glooctl" codelang="shell">}}
+glooctl create vs --name http-echo --namespace gloo-system
+glooctl add route --path-prefix /echo --dest-name default-http-echo-5678
+{{< /tab >}}
+{{< /tabs >}} 
+
+To verify that the Virtual Service works, let's send a request to `/echo`:
+
+```bash
+curl $GATEWAY_URL/echo
+'Hello World!'
+```
+
+#### Deploy an LDAP server
+We also need to deploy an LDAP server to your cluster and configure it with a simple set of users and groups. This 
+information will be used to determine which requests can access the upstream we just defined. 
+
+We have prepared a [**shell script**](setup-ldap.sh) that takes care of setting up the necessary resources. It creates:
+
+1. a `configmap` with the LDAP server bootstrap configuration
+2. a `deployment` running OpenLDAP
+3. a `service` fronting the deployment
+ 
+The script accepts an optional string argument, which determines the namespace in which the resources will be created 
+(`default` if not provided). After you have downloaded the script to your working directory, you can run the following 
+commands to execute it:
+
+```shell
+chmod +x setup-ldap.sh
+./setup-ldap.sh    
+
+No namespace provided, using default namespace
+Creating configmap with LDAP server bootstrap config...
+configmap/ldap created
+Creating LDAP service and deployment...
+deployment.apps/ldap created
+service/ldap created
+```
+
+{{% expand "The details of the script are beyond the scope of this guide; if you are interested, you can inspect them by clicking on this paragraph." %}}
+```bash
+{{< readfile file="gloo_routing/virtual_services/security/ldap/setup-ldap.sh" >}}
```

Comment:
this doesn't render properly, we probably need to make a change similar to this: https://github.com/solo-io/hugo-theme-soloio/pull/2 in the expand tag, and then bump our hugo-theme deps in this repo